src/main/frontend/mixins.cljs

```diff
@@ -87,6 +87,13 @@
                (when (and all-handler (fn? all-handler))
                  (all-handler e key-code)))))))
 
+(defn on-input
+  ([state all-handler target]
+   (listen state (or target js/window) "input"
+           (fn [e]
+             (let [new-text e.event_.data]
```

Comment:
I didn't see a better way to access `e.event_.data`; it doesn't seem exposed in the normal clojure way. LMK if there's a way I should have done this instead.

Reply:
The original codebase here is very awful. If I remember correctly, the `e.event_` is only available under debug mode.
We should break the dependence on debugging member in future.